### ISSUE TITLE
Geographic dot-dropping: Select which point attributes to keep/drop using priority ranking

### ISSUE THREAD

**Issue by marioaks** [mapbox/tippecanoe#842]
I have a dataset of 6M points where each point has a priority value from 0-10. I am using `--cluster-distance=60` in combination with `-r1` in order to cluster/drop points at every zoom level. The goal is to display each of these points on a map using a 60px icon without any other point colliding with it. 

So far, I have not figured out if there is any way to specify which point to keep within a cluster. As far as I can tell, there is no current way of declaring some logic to decide which point should be kept and which should be dropped. It is also my understanding that all attributes will be arbitrarily taken from the first feature in the cluster, but I don't actually know how this _first_ feature is being selected. 

Is there a way to set which point attributes are kept or manipulate the way the _first_ point is chosen? If not, is there some other way to accomplish the task of not only geographically clustering points, but dropping based on a ranking? I have considered using `--accumulate-attribute=id:comma` and selecting the id manually but I'm worried that the accumulated id field will get too large on low zoom levels. 

I don't believe this is a duplicate of https://github.com/mapbox/tippecanoe/issues/351 because they want to drop points "solely" based on rank, but I want to drop points based on geography and identify the remaining point using a rank. 

Thanks!

**Comment by ericfischer** [comment]
Currently the first point in each cluster (in quadkey or Hilbert) order will always win for that cluster. That will tend to be the top-left one but will not always be.

Would the `max` or `min` option to `--accumulate-attribute` be close enough to what you want?

**Comment by marioaks** [comment]
I figured it out eventually! I ended up sorting all the points by priority and giving each point a unique id corresponding to it’s order in the sorted list. Then I generated two separate csv files, one with the new generated ids and the point’s coordinates, and another with the new ids and all the other point metadata. Then I used the accumulate-attribute min option on the sorted ids to get the id with the highest priority for each cluster. Finally I used the tile-join csv option to merge in all the metadata from the second csv file.

in hindsight this was relatively straightforward, but I think it might make sense to either simplify or document this process. Dropping features based on a value (whether it’s points or polygons) seems like something this library should be able to easily do.
